chart/kafka-connector/templates/deployment.yml

```diff
@@ -16,7 +16,7 @@ metadata:
     app.kubernetes.io/part-of: openfaas
     app.kubernetes.io/managed-by: {{ .Release.Service }}
     helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
-  name: kafka-connector
+  name: {{ template "connector.fullname" . }}
```

Comment:
What is this going to output if nobody defines `fullname`?

Reply:
This is using the typical full name helper that many charts have 


```
{{- define "connector.fullname" -}}
{{- if .Values.fullnameOverride -}}
{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
{{- else -}}
{{- $name := default .Chart.Name .Values.nameOverride -}}
{{- if contains $name .Release.Name -}}
{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
{{- else -}}
{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
{{- end -}}
{{- end -}}
{{- end -}}
```

If `fullnameOverride` is set, it completely overrides the deployment name
Otherwise, the name is either _nameOverride_ if it is defined or default to _the chart name_ and then the name is constructed at `releaseName-name` truncated to 63 characters.

Reply:
So we don't need to add the macro above, it's just automatic?